Makefile

```diff
@@ -278,7 +278,7 @@ $(BUILD_IMAGE): bin/calico-typha-$(ARCH) register
 	rm -rf docker-image/bin
 	mkdir -p docker-image/bin
 	cp bin/calico-typha-$(ARCH) docker-image/bin/
-	docker build --pull -t $(BUILD_IMAGE):latest-$(ARCH) --build-arg QEMU_IMAGE=$(CALICO_BUILD) --file ./docker-image/Dockerfile.$(ARCH) docker-image
+	docker build --pull -t $(BUILD_IMAGE):latest-$(ARCH) --build-arg QEMU_IMAGE=$(CALICO_BUILD) --build-arg GIT_VERSION=$(GIT_VERSION) --file ./docker-image/Dockerfile.$(ARCH) .
```

Comment:
Add in GIT_VERSION build arg and change docker context path to `.` (so that we can access the LICENSE).

The `.dockerignore` makes sure we have only the files we want in the docker build context.

Reply:
Would prefer if you didn't change the build path.  Just copy any files you need into the docker-image staging directory before running the build.

Issue is that it sends all the go build caches to the docker daemon, which can be hundreds on MB.